pom.xml

```diff
@@ -116,7 +116,7 @@
     <log4j.version>2.17.2</log4j.version>
     <!-- make sure to update IsolatedClientLoader whenever this version is changed -->
     <hadoop.version>3.3.3</hadoop.version>
-    <protobuf.version>2.5.0</protobuf.version>
+    <protobuf.version>3.21.1</protobuf.version>
```

Comment:
Maybe we should shade and relocation `protobuf-java` in spark to avoid  potential conflicts with other third-party libraries, such as hadoop

Reply:
hadoop hdfs has its own shaded copy now so doesn't care(*); a protobuf upgrade is incompatible with any code compiled against the later version, so must be shaded somehow. 

(*) more specifically, has a new problem, how to safely upgrade that shaded hadoop-thirdparty jar with guava, protobuf etc

Reply:
> a protobuf upgrade is incompatible with any code compiled against the later version, so must be shaded somehow.

@steveloughran what does this mean? could you point to more details?
If I shaded 'com.google.protobuf', in my jar, I can update it anytime, right?